sdk/src/orderSubscriber/OrderSubscriber.ts

```diff
@@ -106,7 +110,18 @@ export class OrderSubscriber {
 					'User',
 					buffer
 				) as UserAccount;
-
+			const newOrders = userAccount.orders.filter(
+				(order) => order.slot.toNumber() === slot
```

Comment:
this will get you all the orders that were placed in the `slot` where your new buffer data is retrieved, i think you want all ' new orders since the order subscriber last had data' ? so something like `(order) => order.slot.toNumber() > slotAndUserAccount.slot` and you'd need to handle the undefined case